motionplan/check.go

```diff
@@ -168,7 +173,7 @@ func checkPlanRelative(
 	expectedCurrentPose := spatialmath.PoseBetweenInverse(remainingArcPose, expectedArcEndInWorld.Pose())
 	errorState := spatialmath.PoseBetween(expectedCurrentPose, currentPoseInWorld.Pose())
 
-	planStartPiF, ok := plan.Path()[0][checkFrame.Name()]
+	planStartPiF, ok := plan.Path()[wayPointIdx][checkFrame.Name()]
```

Comment:
Good question. 
`planStartPiF` is turned into `planStartPoseWorld` which is then passed into `plannerSetupFromMoveRequest`. 

On `main` and within `plannerSetupFromMoveRequest` we would then transform the `solverFrame` off our `seedMap` of type `map[string][]referenceframe.Input`. But, since this `seedMap` contains relative inputs, the transform's result would incorrectly place us in a relative position. We were then able to position ourselves absolutely since we were composing the relative position with `planStartPoseWorld`.

Now suppose that our `wayPointIdx` is `2`. Then, the `planStartPiF` should reflect that.
However, this itself is not entirely correct either.


Since `solverFrame.Transform(seedMap)` will now place us in our absolute position from the get-go, we do not need to compose its output with anything else. Hence, I believe the correct course of action here is to actually remove `planStartPiF`, and `planStartPoseWorld` and instead pass in `currentPoseInWorld` into `plannerSetupFromMoveRequest`

Reply:
The intended use of `planStartPoseWorld` is to place all frames where they were when the plan was created, as this will accurately reflect which things may or may not be in collision, something the current inputs may not do.

As the first element of the plan will generally be all 0 inputs for a relative plan, that should not have placed the frame incorrectly.

Reply:
That said, relying on the first set of inputs to be 0 is bad, and thus this PR will be an improvement. But we will still want to use index 0 of the plan.

Reply:
> as this will accurately reflect which things may or may not be in collision, something the current inputs may not do.

This way my missing mental link. Thank you for explaining :) I am pretty sure I get it now.


I want to ensure that I understand the `"why"` we may not rely on current inputs to correctly inform us about allowed/un-allowed collisions.

Suppose we have just started executing a plan and we are still close to our start pose.
Suppose that we have some transient obstacle named `X` which is positioned on the path, e.g. 2 meters ahead.
If for some reason at this iteration of `CheckPlan` we do not correctly identify collision with `X` then we risk having it being included in our `zeroCg`?